lib/puppet/application/agent.rb

```diff
@@ -39,6 +39,7 @@ def preinit
       :graph => true,
       :fingerprint => false,
       :sourceaddress => nil,
+      :start_time => Time.now,
```

Comment:
Note to self, `preinit` is called early on in the application lifecycle, "The preinit block is the first code to be called in your application, before option parsing, setup or command execution." It won't include any of the time spent loading puppet ruby files, but should be good enough.